python/ml4ir/base/model/scoring/scoring_model.py

```diff
@@ -227,6 +227,10 @@ def final_activation_op(self, scores, metadata_features):
             Tensor object produced by applying the final activation function
             to the scores computed by the model
         """
+        if isinstance(self.loss, dict):
```

Comment:
Why not enforce loss to be a dict? Will it need a lot of refactoring?

Reply:
I think it's doable, I just didn't feel the need to make that change at the moment.